subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -168,6 +197,44 @@ void radio_isr_set(radio_isr_cb_t cb, void *param)
 	irq_enable(RADIO_IRQn);
 }
 
+int radio_init(void)
+{
+#if defined(HAL_RADIO_GPIO_HAVE_PA_PIN) || defined(HAL_RADIO_GPIO_HAVE_LNA_PIN)
+	if (nrfx_gpiote_channel_alloc(&gpiote_palna, &gpiote_ch_palna) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+#endif
+
+#if defined(NRF_GPIO_PDN_PIN)
+	if (nrfx_gpiote_channel_alloc(&gpiote_pdn, &gpiote_ch_pdn) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+#endif
+
+#if defined(NRF_GPIO_CSN_PIN)
+	if (nrfx_gpiote_channel_alloc(&gpiote_csn, &gpiote_ch_csn) != NRFX_SUCCESS) {
+		return -ENOMEM;
+	}
+#endif
```

Comment:
Is there an option to use device tree here? we are now using RAM variables and runtime CPU usage to allocate channels. This will influence startup time, right?